File: api-service/src/modules/errors/http.error.ts
import { AppError } from "./app.error.js";

export { AppError };

export class HttpError extends AppError {
  private constructor(message: string, status: number) {
    super(message, status);
  }

  static NotFoundError(message: string = "Resource not found."): HttpError {
    return new HttpError(message, 404);
  }

  static UnauthorizedError(message: string = "Unauthorized."): HttpError {
    return new HttpError(message, 401);
  }

  static BadRequestError(message: string = "Invalid request."): HttpError {
    return new HttpError(message, 400);
  }

  static TooManyRequestsError(
    message: string = "Too many requests. Please try again later.",
  ): HttpError {
    return new HttpError(message, 429);
  }
}

File: embedding-service/src/modules/errors/http_error.py
from src.modules.errors.app_error import AppError


class HTTPError(AppError):
    @staticmethod
    def not_found(message: str = "Resource not found.") -> "HTTPError":
        return HTTPError(message, 404)

    @staticmethod
    def unauthorized(message: str = "Unauthorized.") -> "HTTPError":
        return HTTPError(message, 401)

    @staticmethod
    def bad_request(message: str = "Invalid request.") -> "HTTPError":
        return HTTPError(message, 400)

    @staticmethod
    def too_many_requests(
        message: str = "Too many requests. Please try again later.",
    ) -> "HTTPError":
        return HTTPError(message, 429)
